darts/explainability/shap_explainer.py

```diff
@@ -0,0 +1,474 @@
+            This is optional, for 2 reasons:
+                - In general we want to keep the training_series of the model and this is the default one,
+                but in case of multiple time series training (global or meta learning) the ForecastingModel doesn't
+                save them. In this case we need to feed a background time series.
+                - We might want to consider a reduced well chosen background in order to reduce computation 
+                time.
+        background_past_covariates
+            A past covariates TimeSeries or list of TimeSeries that the model needs once fitted.
+        background_future_covariates
+            A future covariates TimeSeries or list of TimeSeries that the model needs once fitted.
+        n
+            Optionally, Number of predictions ahead we want to explain. It will provide complete explanation for each 
+            future n steps.
+            If we have existence of output_chunk_length, then n := output_chunk_length automatically.
+            (TODO Not sure it is really needed. Indeed, if there is no output_chunk_length, would be just n=1
+            enough? The only question is do we want to explain at any time step in the future with 
+            autoregressive models. Also, we could explain further than output_chunk_length)
+        past_steps_explain
+            A number of timesteps in the past of which we want to estimate the shap values. If x_t is 
+            our prediction, past_steps_explain = p will take (t-1) ... (t-p) features in the past 
+            (past target, past covariates, future covariates).
+            If we have existence of input_chunk_length (for regression models, the farthest past lag gives the length), 
+            then past_steps_explain := input_chunk_length.
+        
+        TODO
+            Optional De-trend  if the timeseries is not stationary.
+            There would be 1) a stationarity test and 2) a de-trend methodology for the target. It can be for
+            example target - moving_average(input_chunk_length).
+
+        """
+        super().__init__(
+            model,
+            background_series,
+            background_past_covariates,
+            background_future_covariates,
+            n,
+            past_steps_explained
+        )
+
+      
+        self.explainers = RegressionShapExplainers(
+            self.model,
+            self.background_series,
+            self.background_past_covariates,
+            self.background_future_covariates
+        )
+
+    def explain_from_input(
+        self,
+        foreground_series: TimeSeries, 
+        foreground_past_covariates: Optional[TimeSeries], 
+        foreground_future_covariates: Optional[TimeSeries], 
+        horizons: Optional[Sequence[int]] = None, 
+        target_names: Optional[Sequence[str]] = None
+        ) -> Sequence[Sequence[TimeSeries]]:
```

Comment:
I think here it would be nice to explain how the model relates to this foreground series - i.e., that some "features" (i.e., "input sub-slices") are extracted from this series in a way that is similar to how the training samples are obtained, and the shap values are computed on those samples as they are shifted over time. And explain the result type also (btw it doesn't seem to be a `Sequence[Sequence[TimeSeries]]`).